tests/XrdCl/XrdClSocket.cc

```diff
@@ -0,0 +1,301 @@
+//------------------------------------------------------------------------------
+// Copyright (c) 2011-2012 by European Organization for Nuclear Research (CERN)
+// Author: Lukasz Janyst <ljanyst@cern.ch>
+//------------------------------------------------------------------------------
+// XRootD is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// XRootD is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with XRootD.  If not, see <http://www.gnu.org/licenses/>.
+//------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+#include <cstdlib>
+#include <ctime>
+#include <random>
+#include <chrono>
+#include "Server.hh"
+#include "Utils.hh"
+#include "TestEnv.hh"
+#include "XrdCl/XrdClSocket.hh"
+#include "XrdCl/XrdClUtils.hh"
```

Comment:
Please re-order header includes trying to follow something like https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes. That is, include the `.hh` corresponding to the `.cc` first, then local headers, then system headers.